### DISCUSSION TITLE
IDO SOM 2D01 [SSD201] Can not start u-boot

### DISCUSSION THREAD

**Discussion by BoicencoOleg** [linux-chenxing/linux-chenxing.org#69] [General]
Hi to all!
I have started to work with the IDO SOM2D01 module. I got the SDK and the ISP tool.

I did the steps as described in the following manual: http://doc.industio.com/docs/ssd20x_manual/ssd20x_manual-1cradq5hh4qn2
Finally, I obtained the files: GCIS.bin; IPL.bin; IPL_CUST.bin; u-boot_spinand.xz.img.bin
I wrote the images to the NAND flash at the following addresses:
GCIS.bin -> 0x000000
IPL.bin -> 0x140000
IPL_CUST.bin -> 0x200000
u-boot_spinand.xz.img.bin -> 0x2C0000

After successfully flashing and restarting the chip I observe printing the following messages:

`IPL g76cd9eb
D-1d

HW Reset
miupll_166MHz
miu_bw_set
utmi_1_init done
utmi_2_init done
utmi_3_init done
usbpll init done......
cpupll init done
SPI 54M
clk_init done 
P1 USB_rterm trim=0x0001
P1 USB_HS_TX_CURRENT trim=0x0002
P2 USB_rterm trim=0x0001
P2 USB_HS_TX_CURRENT trim=0x0002
P3 USB_rterm trim=0x0001
P3 USB_HS_TX_CURRENT trim=0x0002
PM_vol_bgap trim=0x0000
GCR_SAR_DATA trim=0x0191
ETH 10T output swing trim=0x0010
ETH 100T output swing trim=0x0010
ETH RX input impedance trim=0x0000
ETH TX output impedance trim=0x0001
MIPI_HS_RTERM trim=0x0001
MIPI_LP_RTERM trim=0x0000

64MB

BIST0_0001-OK

Enable MMU and CACHE
Load IPL_CUST from SPINAND
QUAD MODE ENABLE
unable to find IDX for part type:0001
[I]m7
`

As far as I understand from the log, the IPL was running but the IPL_CUST has not been running. Accordingly, u-but was not  launched (called/running).

After some investigation, I observed that the second 4 bytes of my IPL_CUST contained symbols "IPLN", not "IPLC" as described here: http://linux-chenxing.org/blobs/ipl.html

The second my step was cloning buildroot from: https://github.com/fifteenhex/buildroot_idosom2d01. 
But unfortunately, the building (make) failed. 
However 'make' did some output images. I found u-boot-spl.bin in the output directory. And this file contains the correct second 4 bytes ("IPLC"). 
I tried to flash this image at address 0x140000, but ROM-BOOT says that "header has not been recognized" (something in this sense). Then I tried to flash IPL.bin at address 0x140000 and u-boot-spl.bin at 0x200000 the output messages have not changed (it is the same as above) - IPL running, but SPL did not run.

Please, give me any ideas on how to run u-boot in my case?! Thanks in advance!!
Maybe someone can share the working images?

P.S. Attaching all the files I used
[Vendor_boot_files.zip](https://github.com/linux-chenxing/linux-chenxing.org/files/9351587/Vendor_boot_files.zip)
[git_buildroot_boot_files.zip](https://github.com/linux-chenxing/linux-chenxing.org/files/9351593/git_buildroot_boot_files.zip)
.

**Comment by BoicencoOleg** [comment]
In addition to the previous post:
As I wrote above, I was unable to make the [buildroot_idosom2d01] (https://github.com/fifteenhex/buildroot_idosom2d01).
Attaching the files with make log and make configurations. A question: does someone knows why make failed in my case?

Thanks in advance!

[buildroot_idosom2d01_make_log_and_config.zip](https://github.com/linux-chenxing/linux-chenxing.org/files/9364233/buildroot_idosom2d01_make_log_and_config.zip)
!

**Comment by fifteenhex** [comment]
Can you try the binaries here:
https://github.com/fifteenhex/buildroot_idosom2d01/releases/tag/build_20220404_00

Pretty sure these worked.

**Comment by BoicencoOleg** [comment]
Thank you, dear friend!! This works!
